api/client/client.go

```diff
@@ -0,0 +1,249 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/base64"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/api/grpc/disperser"
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigenda/core/auth"
+	"github.com/Layr-Labs/eigenda/encoding"
+	"github.com/Layr-Labs/eigenda/encoding/utils/codec"
+	"github.com/ethereum/go-ethereum/log"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+type BlobEncodingVersion byte
+
+var NoIFFT BlobEncodingVersion = 0x00
+
+type IEigenDAClient interface {
+	GetBlob(ctx context.Context, BatchHeaderHash []byte, BlobIndex uint32) ([]byte, error)
+	PutBlob(ctx context.Context, txData []byte) (*Cert, error)
+}
+
+type EigenDAClient struct {
+	Config
+
+	Log log.Logger
+
+	client disperser.DisperserClient
```

Comment:
any reason why we aren't mocking the client and running assertions on the put loop failure cases? so far there's a test to verify correct dispersal/retrieval flows with testnet but nothing pertaining to other interactions

Reply:
No reason except that mocking out the entire API server seems beyond the scope of this PR.